app_dart/test/request_handlers/vacuum_github_commits_test.dart

```diff
@@ -98,6 +97,10 @@ void main() {
       when(githubService.github.repositories).thenReturn(repositories);
     });
 
+    tearDown(() {
+      db.values.clear();
```

Comment:
`db` is being set to a new `FakeDatastoreDB` in `setUp`; why is this necessary?

Reply:
I'm not sure. I've noticed similar weirdness with other async tests before where we need to add tearDown logic.

The core issue was commits from other tests were in this test's `db`

Reply:
How is that possible?

If somehow tests are running in parallel with cross-contaminated `db` instances then this doesn't seem like a safe fix, because you could just as easily be clearing the commits from a different test while it was executing.

Reply:
I was curious, so I tried running just this one test locally, and it still fails exactly the same way, so I don't see how it can be caused by other tests.

Reply:
Just replacing the magic number `4` with `githubCommits.length` (which is 3) as you did below fixes the test, without this change. If the core issue is that `db` was wrong, why are the expectations changing?

Reply:
I've removed the tearDown. I'm on my linux workstation today and I'm not able to reproduce.